### PR TITLE
Update Data as a Product page

### LINKED ISSUE #1256: Refresh architecture diagrams: Run data product
Confluence page https://dsdmoj.atlassian.net/wiki/spaces/DataPlatform/pages/4353818679/Run+Data+Product+in+Data+platform+POC

Update the diagrams to reflect the MVP architecture and hand it off to Haymon

### PR DESCRIPTION
**Pull request by tom-webber**
Refresh the Data as a Product page with the current state of the process.
- Add API overview and vision
- replace architecture and data-flow diagrams

resolves #1256

### CODE REVIEW

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb [hunk 1]
```diff
@@ -1,48 +1,65 @@
...
 
-<strong>The architecture will be built in the `data-platform-development`, a sandbox account within MoJ's Modernisation Platform, which is deployed using the [modernisation-platform-environments](https://github.com/ministryofjustice/modernisation-platform-environments) repository</strong>
+We want to design the Data Product management process around a user-facing API.
+This API will allow users to register, update, and retrieve information about a Data Product.
 
-## Data Producer Configuration Flow
+The data sent to this API as part of registering or updating a data product will flow through standard Data Lake zones: Landing -> Raw -> Curated.
 
-![Data platform general diagram](../images/data-platform-configuration-flow.png "Data platform configuration Flow")
+Each Data Product will require a [specification](https://github.com/ministryofjustice/data-platform-products/blob/main/data-products/_docs/product-specification.md) (still under refinement) alongside [other definitions](https://github.com/ministryofjustice/data-platform-products/tree/main/data-products#defining-a-data-product).
```
Comment (MatMoore) [inline docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb]: Is this up to date? I haven't seen any of these yaml files before, just the example metadata file (a single JSON)

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb [hunk 2]
```diff
@@ -1,48 +1,65 @@
...
 
-**For now** - Data product configuration will be sent to the Data Platform S3 bucket via [GitHub Actions](https://github.com/ministryofjustice/data-platform-products/blob/main/.github/workflows/data-product-example-2-poc.yaml).
+Data Products:
```
Comment (MatMoore) [inline docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb:15]: Maybe this para can be moved further up - it makes sense to define what we mean by a data product before talking about what the API does

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb [hunk 3]
```diff
@@ -1,48 +1,65 @@
...
 
-The data product configuration (Zip file) will be stored in the following structure:
-> `${DATA_PLATFORM_PRODUCTS}/code/${DATA_PLATFORM_PRODUCT_NAME}/${ZIPFILE}.zip`
+* May be multi-component, e.g. consisting of multiple tables
+* Can consist of structured or unstructred data (or a combination)
+* Are controlled and updated by Data Owners, Data Stewards, and Data Custodians.
```
Comment (MatMoore) [inline docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb]: These terms should definitely be defined on first use (I'm not sure of the exact meanings myself, and they aren't defined in our [team glossary](https://docs.google.com/spreadsheets/d/1O1EjO96lyqDbyuzFWU8SNBky3b7tKM8dccZVS7-_t9g/edit#gid=1105700365))

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb [hunk 4]
```diff
@@ -1,48 +1,65 @@
...
+This API is backed by containerised lambdas that carry out each step of the event-driven data processing pipeline.
+Data will first go to a ‘landing’ bucket ahead of data validation.
+It is then moved to a ‘raw history’ bucket folder to retain a copy of the data.
+The arrival of structured data in the raw history folder triggers a ‘curation’ lambda
+which standardises the file type and data format before placing the data in the finalised ‘curated’ bucket folder.
+This curated data is the data that is surfaced as the data product, and is registered to the aws Glue catalogue.
 
-The data will be sent via an api, and will land into the data-platform-bucket
-> `${DATA_PLATFORM_PRODUCTS}/${DATA_PLATFORM_PRODUCT_NAME}/rawdata/extraction_timestamp/${SOMEDATA}.csv
+As part of the data flow, Data Product metadata must be passed as part of the request to register or amend a Data Product.
```
Comment (MatMoore) [inline docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb]:
```suggestion
As part of the request to register or amend a Data Product, the API user must provide Data Product metadata.
```

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb [hunk 5]
```diff
@@ -54,28 +71,63 @@ The following are the assumptions and risks. This will be mitigated eventually.
...
+### Athena vs. Glue
+Athena is cheaper than using Glue for compute as you are charged for amount of data scanned per query ($5 per TB).
+Cost of lambda execution is also relatively inexpensive at $0.0000083 per second with 512 Mb memory.
 
-2. There is only one bucket (Data Platform bucket). Everything else will be sub folders of the same bucket
+Initial testing has shown Athena as compute to be a lot quicker than using a Glue Job for moderately sized data.
+Further benchmarking is required as scale and complexity increases.
+
+### Additional developments/ improvements
+
```
Comment (MatMoore) [inline docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb]: Who is the audience for this section? Is it for the team or wider stakeholders? I'm just wondering this duplicating info that should live in the epics/roadmap?

**Review by MatMoore — DISMISSED** [review]
Looks really good, and those diagrams look accurate to me. How do we access the source for the pngs if we need to edit them? Are you able to link this in the readme?

**Comment by tom-webber** [comment]
> How do we access the source for the pngs if we need to edit them? Are you able to link this in the readme?

I can put the excalidraw json source in the images folder. Might need a reference somewhere in a CONTRIBUTING file to clarify this (not 100% sure where this should go)

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb (hunk 1 above)
Comment (tom-webber) [reply docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb]: I'm not sure. @jemnery do we know what the integration is/should be around the API and these specifications?

**Comment by bagg3rs** [comment]
> > How do we access the source for the pngs if we need to edit them? Are you able to link this in the readme?
> 
> I can put the excalidraw json source in the images folder. Might need a reference somewhere in a CONTRIBUTING file to clarify this (not 100% sure where this should go)

@tom-webber Just add a `.png` to the `.excalidraw` file and you can edit them via vscode with extension or with excali web. 
https://technical-documentation.data-platform.service.justice.gov.uk/documentation/diagrams/architecture-diagrams.html#contributing-and-updating

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb (hunk 3 above)
Comment (tom-webber) [reply docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb]: @jemnery whereabouts can I find references to what we mean in Data Platform by these terms?

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb (hunk 5 above)
Comment (tom-webber) [reply docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb]: Removed - now this only resides in the internal version of this document (on confluence)

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb [hunk 6]
```diff
@@ -1,48 +1,65 @@
...
-<strong>The architecture will be built in the `data-platform-development`, a sandbox account within MoJ's Modernisation Platform, which is deployed using the [modernisation-platform-environments](https://github.com/ministryofjustice/modernisation-platform-environments) repository</strong>
+Each Data Product will require a [specification](https://github.com/ministryofjustice/data-platform-products/blob/main/data-products/_docs/product-specification.md) (still under refinement) alongside [other defined pieces of descriptive metadata captured at registration](https://github.com/ministryofjustice/data-platform-products/tree/main/data-products#defining-a-data-product).
 
-## Data Producer Configuration Flow
+Data Products:
 
-![Data platform general diagram](../images/data-platform-configuration-flow.png "Data platform configuration Flow")
+* May be multi-component, e.g. consisting of multiple tables
+* Can consist of structured or unstructured data (or a combination)
+* Are controlled and updated by Data Owners, Data Stewards, and Data Custodians.
```
Comment (jemnery) [inline docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb]: Suggest just "**Are controlled and updated by Data Owners**" for now, as you're right, I'm fairly sure we haven't defined the Data Steward and Data Custodian roles. In fact those definitions are really up to the DIP team, so let's not create a blocker waiting for those.

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb [hunk 7]
```diff
@@ -1,48 +1,65 @@
...
+We want to design the Data Product management process around a user-facing API.
+This API will allow users to register, update, and retrieve information about a Data Product.
 
-The data product configuration (Zip file) will be stored in the following structure:
-> `${DATA_PLATFORM_PRODUCTS}/code/${DATA_PLATFORM_PRODUCT_NAME}/${ZIPFILE}.zip`
+The data sent to this API as part of registering or updating a data product will flow through standard Data Lake zones: Landing -> Raw -> Curated.
 
-Event Bridge connection to bucket should be enabled. Once the file lands in the landing zone, S3 sends an event to Amazon Event Bridge whenever data events happen in the bucket. We will use Event Bridge rules to route events Object Created `PutObject` events to Lambda.
+> Implementation: The architecture will be built in the `data-platform-development` aws account, a sandbox account within [MoJ's Modernisation Platform](https://user-guide.modernisation-platform.service.justice.gov.uk/#modernisation-platform), which is deployed within the [modernisation-platform-environments](https://github.com/ministryofjustice/modernisation-platform-environments) repository
```
Comment (jemnery) [inline docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb]: Let's not be environment-specific here (sandbox). Suggest something like simply this:

The infrastructure will be built in the Data Platform's [Modernisation Platform](https://user-guide.modernisation-platform.service.justice.gov.uk/#modernisation-platform) [environment(s)](https://github.com/ministryofjustice/modernisation-platform-environments/tree/main/terraform/environments/data-platform)

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb (hunk 3 above)
Comment (jemnery) [reply docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb]: You're right, I'm fairly sure we haven't defined the Data Steward and Data Custodian roles. In fact those definitions are really up to the DIP team, so let's not create a blocker waiting for those. There's a suggested edit in the diff.

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb (hunk 1 above)
Comment (jemnery) [reply docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb]: Those YAML specs are still valid in that they contain a decent hypothesis of what metadata we want to capture, but they date from a while back when we thought data product creation could be raising a PR against the "products" repo (like creating a Cloud Platform namespace). So ignore the format, just take it as warning the metadata specs will evolve from here.

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb [hunk 8]
```diff
@@ -54,28 +71,41 @@ The following are the assumptions and risks. This will be mitigated eventually.
...
+Various reporting tools can be attached later.
 
-##### Event Bridge
+#### Use of aws services
 
-We are not choosing Event Bridge to directly call Lambda for for following reasons:
+##### API Gateway
 
-1. Event Bridge to Glue needs a workflow setup. We cannot extract the full event in Glue script, only the name can be extracted. With Lambda, the whole event can be passed and we can extract the information needed to run the script.
+API gateway is used to start the data ingestion process.
```
Comment (jemnery) [inline docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb]: API Gateway is used to host and manage our API endpoints, so a user will interact with the API to initiate the product registration and data ingestion processes.

File: docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb [hunk 9]
```diff
@@ -54,28 +71,41 @@ The following are the assumptions and risks. This will be mitigated eventually.
...
+Athena is cheaper than using Glue for compute as you are charged for amount of data scanned per query ($5 per TB).
+Cost of lambda execution is also relatively inexpensive at $0.0000083 per second with 512 Mb memory.
+
+Initial testing has shown Athena as compute to be a lot quicker than using a Glue Job for moderately sized data.
+Further benchmarking is required as scale and complexity increases.
+
+### Notes
 
-##### Reporting tool
+#### Reporting tool
```
Comment (jemnery) [inline docs/source/documentation/platform/infrastructure/run-data-product-in-platform.html.md.erb]: I suggest deleting the "Reporting Tool" section from here as it's not adding much and not really relevant to product creation / data ingestion

**Review by jemnery — CHANGES_REQUESTED** [review]
4 minor edit suggestions inline

**Review by jemnery — APPROVED** [review]

**Pull request merged**